src/gb_parser.yrl

```diff
@@ -6,72 +6,57 @@ dot lbracket rbracket lparen rparen
+template template_exprs template_expr tag_attrs tag_attr attr_name var_value var_expr var_ops var_op.
 
 Rootsymbol template.
 
-Expect 2.
+Nonassoc 100 expr_name.
+Nonassoc 200 tag.
+Nonassoc 300 body_tag.
 
 template ->
   template_exprs : ensure_list('$1').
 
 template_exprs ->
-  text : {text, value_from('$1')}.
-template_exprs ->
-  expr_name : unknown_tag('$1').
+  template_expr template_exprs : combine('$1', '$2').
 template_exprs ->
+  '$empty' : nil.
+
+template_expr ->
+  text : {text, value_from('$1')}.
+template_expr ->
+  newline : make_newline('$1').
+template_expr ->
+  end_of_collapsible_body_tag : nil.
+template_expr ->
   expr_name tag_attrs : unknown_tag('$1').
-template_exprs ->
+template_expr ->
   expr_name tag_attrs template_exprs expr_end : unknown_tag('$1').
-template_exprs ->
-  tag : make_tag('$1').
-template_exprs ->
+template_expr ->
   tag tag_attrs : make_tag('$1', '$2').
-template_exprs ->
-  body_tag expr_end : make_tag('$1', [], []).
-template_exprs ->
-  body_tag tag_attrs expr_end : make_tag('$1', '$2', []).
-template_exprs ->
-  body_tag template_exprs expr_end : make_tag('$1', [], '$2').
-template_exprs ->
+template_expr ->
   body_tag tag_attrs template_exprs expr_end : make_tag('$1', '$2', '$3').
-template_exprs ->
+template_expr ->
+  newline body_tag tag_attrs template_exprs expr_end newline :
+    combine(make_newline('$1'), make_tag('$2', '$3', maybe_combine_newline('$2', strip_newlines('$4'), make_newline('$6')))).
+template_expr ->
+  end_of_collapsible_body_tag body_tag tag_attrs template_exprs expr_end newline :
+    make_tag('$2', '$3', maybe_combine_newline('$2', strip_newlines('$4'), make_newline('$6'))).
+template_expr ->
   var_value : '$1'.
-template_exprs ->
+template_expr ->
   eol : eol.
-template_exprs ->
-  text template_exprs : combine({text, value_from('$1')}, '$2').
-template_exprs ->
-  expr_name template_exprs : unknown_tag('$1').
-template_exprs ->
-  expr_name tag_attrs template_exprs : unknown_tag('$1').
-template_exprs ->
-  expr_name tag_attrs template_exprs expr_end template_exprs : unknown_tag('$1').
-template_exprs ->
-  tag template_exprs : combine(make_tag('$1'), '$2').
-template_exprs ->
-  tag tag_attrs template_exprs : combine(make_tag('$1', '$2'), '$3').
-template_exprs ->
-  body_tag expr_end template_exprs : combine(make_tag('$1', [], []), maybe_drop_leading_eol('$1', '$3')).
-template_exprs ->
-  body_tag tag_attrs expr_end template_exprs : combine(make_tag('$1', '$2', []), maybe_drop_leading_eol('$1', '$4')).
-template_exprs ->
-  body_tag template_exprs expr_end template_exprs : combine(make_tag('$1', [], '$2'), maybe_drop_leading_eol('$1', '$4')).
-template_exprs ->
-  body_tag tag_attrs template_exprs expr_end template_exprs : combine(make_tag('$1', '$2', '$3'), maybe_drop_leading_eol('$1', '$5')).
-template_exprs ->
-  var_value template_exprs : combine('$1', '$2').
-template_exprs ->
-  eol template_exprs : combine(eol, '$2').
```

Comment:
Wheeeee. Was able to remove this stuff with the addition of the `'$empty'` rule above and a couple of newline related rules.